sbb_binarize/ocrd_cli.py

```diff
@@ -0,0 +1,129 @@
+
+    def process(self):
+        """
+        Binarize with sbb_binarization
+        """
+        LOG = getLogger('processor.SbbBinarize')
+        assert_file_grp_cardinality(self.input_file_grp, 1)
+        assert_file_grp_cardinality(self.output_file_grp, 1)
+
+        oplevel = self.parameter['operation_level']
+        self.use_patches = self.parameter['patches'] # pylint: disable=attribute-defined-outside-init
+        self.model_path = self.parameter['model'] # pylint: disable=attribute-defined-outside-init
+
+        for n, input_file in enumerate(self.input_files):
+            file_id = make_file_id(input_file, self.output_file_grp)
+            page_id = input_file.pageId or input_file.ID
+            LOG.info("INPUT FILE %i / %s", n, page_id)
+            pcgts = page_from_file(self.workspace.download_file(input_file))
+            self.add_metadata(pcgts)
+            pcgts.set_pcGtsId(file_id)
+            page = pcgts.get_Page()
+
+            if oplevel == 'page':
+                LOG.info("Binarizing on 'page' level in page '%s'", page_id)
+                page_image, page_xywh, _ = self.workspace.image_from_page(page, page_id)
+                bin_image = self._run_binarizer(page_image)
+                # update METS (add the image file):
+                bin_image_path = self.workspace.save_image_file(bin_image,
+                        file_id + '.IMG-BIN',
+                        page_id=page_id,
+                        file_grp=self.output_file_grp)
+                page.add_AlternativeImage(AlternativeImageType(filename=bin_image_path, comment="binarized"))
+
+            else:
+                regions = page.get_AllRegions(['Text', 'Table'])
+                if not regions:
+                    LOG.warning("Page '%s' contains no text/table regions", page_id)
+
+                for region in regions:
+                    region_image, region_xywh = self.workspace.image_from_segment(region, page_image, page_xywh)
+
+                    if oplevel == 'region':
+                        region_image_bin = self._run_binarizer(region_image)
+                        region_image_bin_path = self.workspace.save_image_file(
+                                region_image_bin,
+                                "%s_%s.IMG-BIN" % (file_id, region.id),
+                                page_id=page_id,
+                                file_grp=self.output_file_grp)
+                        region.add_AlternativeImage(
+                            AlternativeImageType(filename=region_image_bin_path, comments='binarized'))
+
+                    elif oplevel == 'line':
+                        lines = region.get_TextLine()
+                        if not lines:
+                            LOG.warning("Page '%s' region '%s' contains no text lines", page_id, region.id)
+                        for line in lines:
+                            line_image, line_xywh = self.workspace.image_from_segment(line, page_image, page_xywh)
```

Comment:
```suggestion
                            line_image, line_xywh = self.workspace.image_from_segment(line, page_image, page_xywh, feature_filter='binarized')
```
(see above)

Reply:
No, wait, this should actually use the region images for reference (because they might have been deskewed independently):
```suggestion
                            line_image, line_xywh = self.workspace.image_from_segment(line, region_image, region_xywh, feature_filter='binarized')
```